File: backend/Models/PluginSettings.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class PluginSettings
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public virtual IList<DiscordChannel> IgnoredChannels { get; set; }
        public virtual IList<DiscordRole> AllowedRoles { get; set; }
        public ModerationAction ModerationAction { get; set; }
        public virtual Guild Guild { get; set; }
    }
}

File: frontend/src/app/_models/plugin-settings.ts
import {Guild} from './guild';
import {DiscordChannel} from './discord-channel';
import {DiscordRole} from './discord-role';
import {ModerationAction} from './moderation-action';

export interface PluginSettings {
  id: string;
  ignoredChannels: DiscordChannel[];
  allowedRoles: DiscordRole[];
  moderationAction: ModerationAction;
  guild: Guild;
}
